astropy/modeling/core.py

```diff
@@ -818,6 +820,10 @@ def __call__(self, *inputs, **kwargs):
                     outputs = np.asarray(result[0])
                 else:
                     outputs = [np.asarray(r) for r in result]
+                # If the valid results is a quantity, ensure unit information
+                #  is retained.
+                if valid_result_unit is not None:
+                    outputs = Quantity(outputs, valid_result_unit)
```

Comment:
I think it would be good to add ``copy=False`` here to avoid unnecessary copies of the data. You might also be able to use ``outputs = outputs << valid_result_unit``